tests/Client/UpdaterTest.php

```diff
@@ -302,6 +303,18 @@ public function providerRefreshException()
                     'targets' => 3,
                 ],
             ],
+            [
+                '5.targets.json',
+                ['signed', 'version'],
+                6,
+                new MetadataException("Expected targets version 5 does not match actual version 6."),
+                [
+                    'root' => 5,
+                    'timestamp' => 5,
+                    'snapshot' => 5,
+                    'targets' => 3,
+                ],
```

Comment:
Is there a way to avoid duplicating test condition data in the exception? Could we carry over expected/actual data in a more modular fashion?

Reply:
I don't follow what you mean here. Have the exception here in the test case allows using the same `testRefreshException()` test method for any cases where an exception should occur based on change to specific repo file.

I am up for changing this to be cleaner but since it is existing data provider I would rather do it in separate PR. Could be before or after this 1.

Reply:
Okay. I'll switch my review to "approved" based on this. I think I misunderstood how this was integrating with the rest of the code.